### PR TITLE
added coinbase wallet option

### PR DESCRIPTION
**Pull request by calvintoes**
Added `@web3-react/walletlink-connector` package to add Coinbase Wallet option. Only Coinbase prompt pops up when you click Coinbase wallet. Metamask **and** Coinbase prompt pops up when you click Metamask. 

Walletlink connector implementation was borrowed from the way Uniswap does their connections. 

Also tested on Brave and it works out of the box.

---

If you made a contract change, make sure to complete the checklist below before merging it in master.

Refer to our [documentation](https://github.com/OriginProtocol/security) for more details about contract security best practices.

Contract change checklist:
  - [ ] Code reviewed by 2 reviewers. See [template](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review.md) and [example](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review-Example.md).
  - [ ] Unit tests pass
  - [ ] Slither tests pass with no warning
  - [ ] Echidna tests pass if PR includes changes to OUSD contract (not automated, run manually on local)

### CODE REVIEW

**Comment by micahalcorn** [comment]
We shouldn't need to introduce WalletLink for this, should we? What do we gain by doing that?

**Comment by calvintoes** [comment]
@micahalcorn I noticed that with WalletLink, only the Coinbase prompt pops up when you connect a Coinbase wallet. Thought that UX was better than using the same connector as MM and having both prompt pop up for connecting MM and/or Coinbase.

Another UX benefit with WalletLink, if the user tries to connect with Coinbase and doesn't have the extension installed, it will pop up a Coinbase modal asking to install the extension or scan QR code with phone. 
If we used the MM connector instead and the user doesn't have Coinbase extension, then it'll pop up a MM prompt instead (assuming user has MM extension installed), which doesn't make sense to me.

**Comment by micahalcorn** [comment]
> @micahalcorn I noticed that with WalletLink, only the Coinbase prompt pops up when you connect a Coinbase wallet. Thought that UX was better than using the same connector as MM and having both prompt pop up for connecting MM and/or Coinbase.
> 
> Another UX benefit with WalletLink, if the user tries to connect with Coinbase and doesn't have the extension installed, it will pop up a Coinbase modal asking to install the extension or scan QR code with phone. If we used the MM connector instead and the user doesn't have Coinbase extension, then it'll pop up a MM prompt instead (assuming user has MM extension installed), which doesn't make sense to me.

Okay, as you point out, it still gives you redundant popups when clicking MetaMask, so I'm not sure if it's worth the additional dependency or not. I'll defer to @sparrowDom and @tomlinton. It would also be good to confirm that it plays nicely with WalletConnect.

**Comment by sparrowDom** [comment]
I might be missing what was the goal with this PR. The Coinbase wallet connects nicely to what is currently in production: 
Screenshot_20211130-133745

**Comment by micahalcorn** [comment]
> I might be missing what was the goal with this PR. The Coinbase wallet connects nicely to what is currently in production:

See https://github.com/OriginProtocol/origin-dollar/issues/634 🙂 Users were complaining because they thought we didn't support it. If you're in a desktop environment, you have to explicitly click a button and they didn't know to click the MetaMask option. So I was proposing that we just add a button, particularly now that Coinbase Wallet has a browser extension.

**Review by sparrowDom — COMMENTED** [review]
Aside 2 minor comments I think it is great the way you did it. I've checked this brach-es build size and current master build size and it is almost identical. So the extra packages don't pose a thread to enlarge the website size too much. 

And we already use other connectors from `@web3-reach` repo, so aside from non problematic build size I see no problems having those extra QR-code features. 

great work 👍

**Comment by sparrowDom** [comment]
> > I might be missing what was the goal with this PR. The Coinbase wallet connects nicely to what is currently in production:
> 
> See #634 🙂 Users were complaining because they thought we didn't support it. If you're in a desktop environment, you have to explicitly click a button and they didn't know to click the MetaMask option. So I was proposing that we just add a button, particularly now that Coinbase Wallet has a browser extension.

Thanks, I should have figured that out :)

File: package.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "dependencies": {}
```
Comment (sparrowDom) [inline package.json:2]: this can probably go out

File: dapp/src/utils/connectors.js [hunk 2]
```diff
@@ -52,5 +61,37 @@ export const connectorNameIconMap = {
   WalletConnect: 'walletconnect-icon.svg',
 }
 
+export const connectorsByName = {
```
Comment (sparrowDom) [inline dapp/src/utils/connectors.js:64]: This is probably a result of a bad merge?

**Review by sparrowDom — COMMENTED** [review]
A couple of minor comments inline

File: dapp/src/utils/connectors.js (hunk 2 above)
Comment (calvintoes) [reply dapp/src/utils/connectors.js:64]: What should be removed/edited here?

**Comment by tomlinton** [comment]
Thanks for the fixes @calvintoes. Looks good to me.

The dependencies of walletlink are a bit rough, preact and rxjs? I wonder why it requires that.

File: dapp/src/utils/connectors.js (hunk 2 above)
Comment (sparrowDom) [reply dapp/src/utils/connectors.js:64]: The `connectorsByName` has just been refactored to `connectorNameIconMap` while you are working on this PR: 
https://github.com/OriginProtocol/origin-dollar/blob/master/dapp/src/utils/connectors.js#L48-L53

**Comment by franckc** [comment]
Closing this PR since @calvintoes mentioned to me he won't be able to bring it to the finish line...

**Pull request closed without merging**